### PR TITLE
[KYUUBI #4216] Support to transfer client version for kyuubi hive jdbc and rest client sdk

### LINKED ISSUE #4216: [Improvement] Record the client version both for interactive and batch session
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Record the client version both for interactive and batch session.
- for interactive session
  - transfer the kyuubi-hive-jdbc version when opening connection.
- for batch session
   -  transfer the kyuubi-rest-client version when creating batch request

So that we can check the client version in KyuubiSessionEvent(conf).

If customer call kyuubi service with non-kyuubi-hive-jdbc or native restful way, leave it as empty. 
### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

To close #4216

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
cc @pan3793  @lightning-L

File: kyuubi-hive-jdbc/pom.xml [hunk 1]
```diff
@@ -171,6 +171,14 @@
     </dependencies>
 
     <build>
+
+        <resources>
+            <resource>
+                <filtering>true</filtering>
+                <directory>src/main/resources</directory>
+            </resource>
+        </resources>
```
Comment (pan3793) [inline kyuubi-hive-jdbc/pom.xml:180]: we need it? `src/main/resources` should be the default location for resources.

Comment (turboFei) [reply kyuubi-hive-jdbc/pom.xml:180]: without this:
image

Comment (turboFei) [reply kyuubi-hive-jdbc/pom.xml:180]: with this:

image

**Comment by turboFei** [comment]
image


The ut passed locally, I do not know why it failed on GA, it does not make senses.

File: kyuubi-hive-jdbc/pom.xml (hunk 1 above)
Comment (bowenliang123) [reply kyuubi-hive-jdbc/pom.xml:180]: Where and how (what goal or execution) this placeholder is filled?

Comment (turboFei) [reply kyuubi-hive-jdbc/pom.xml:180]: Not sure

**Comment by turboFei** [comment]
move the UT to perConnectionSuite.

**Review by bowenliang123 — APPROVED** [review]
LGTM

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
we can add a udf `kyuubi_client_version()` in the followup PR.

**Comment by turboFei** [comment]
thanks, merging to master and 1.7

**Comment by turboFei** [comment]
> we can add a udf kyuubi_client_version() in the followup PR.


cc @lightning-L

**Pull request closed without merging**